src/STM32Ethernet.h

```diff
@@ -12,17 +12,18 @@ class EthernetClass {
   IPAddress _dnsServerAddress;
   DhcpClass* _dhcp;
 public:
-  // Initialise the Ethernet shield to use the provided MAC address and gain the rest of the
+  // Initialise the Ethernet with the internal provided MAC address and gain the rest of the
   // configuration through DHCP.
   // Returns 0 if the DHCP configuration failed, and 1 if it succeeded
-  int begin(uint8_t *mac_address, unsigned long timeout = 60000, unsigned long responseTimeout = 4000);
-  void begin(uint8_t *mac_address, IPAddress local_ip);
-  void begin(uint8_t *mac_address, IPAddress local_ip, IPAddress dns_server);
-  void begin(uint8_t *mac_address, IPAddress local_ip, IPAddress dns_server, IPAddress gateway);
-  void begin(uint8_t *mac_address, IPAddress local_ip, IPAddress dns_server, IPAddress gateway, IPAddress subnet);
+  int begin(unsigned long timeout = 60000, unsigned long responseTimeout = 4000);
```

Comment:
I think it should be fine to kept backward compatibility and also Arduino API.
So they should be kept and just call the new by setting the private mac address member